### DISCUSSION TITLE
How to impl ToSql for Serialized struct using serde_json in diesel 2.0

### DISCUSSION THREAD

**Discussion by lisiur** [diesel-rs/diesel#3573] [Q&A] [answered]
```rust
#[derive(AsExpression, FromSqlRow, Debug)]
#[diesel(sql_type = Text)]
pub struct JsonWrapper<T: serde::de::DeserializeOwned + serde::Serialize + fmt::Debug>(pub T);

impl<T: serde::de::DeserializeOwned + serde::Serialize + fmt::Debug> FromSql<Text, Sqlite>
    for JsonWrapper<T>
{
    fn from_sql(bytes: diesel::backend::RawValue<'_, Sqlite>) -> diesel::deserialize::Result<Self> {
        let bytes = <Vec<u8>>::from_sql(bytes)?;
        let inner = serde_json::from_slice::<T>(&bytes)?;
        Ok(JsonWrapper(inner))
    }
}

impl<T: serde::de::DeserializeOwned + serde::Serialize + fmt::Debug> ToSql<Text, Sqlite>
    for JsonWrapper<T>
{
    fn to_sql<'b>(
        &'b self,
        out: &mut diesel::serialize::Output<'b, '_, Sqlite>,
    ) -> diesel::serialize::Result {
        todo!()
    }
}
```

I have tried this:

```rust
<str as ToSql<Text, Sqlite>>::to_sql(&serde_json::to_string(&self.0).unwrap(), out)
```

but the complier complains that: `temporary value dropped while borrowed creates a temporary value which is freed while still in use`.

From what I understand, serde_json does not support returning a reference, so could someone provide assistance with resolving this issue?

**Comment by weiznich** [comment, marked as answer]
See the [relevant section of the migration guide](https://diesel.rs/guides/migration_guide.html#2-0-0-to-sql) or the [documentation of `ToSql`](https://docs.diesel.rs/2.0.x/diesel/serialize/trait.ToSql.html) for information on how to resolve these kind of issues.

**Reply by lisiur** [reply]
Thank you for your reply. I apologize for not thoroughly reading the documentation beforehand.

Here is my workaround for those encountering a similar issue.

```rust
out.set_value(serde_json::to_string(&self.0).unwrap());
Ok(IsNull::No)
```

**Reply by yasamoka** [reply]
Just leaving this here for anyone who has this issue - in my case, `&mut out.reborrow()` was the solution.
